provider/azure/doc.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Package azure implements the Azure provider, registered with
+// environs under the name "azure". The provider implements the
+// [github.com/juju/juju/environs.Environ] interface, which defines
+// methods for provisioning compute, network, and storage resources.
+//
+// Here we describe some key implementation details specific to the Azure provider.
+//
+// # SDK
+//
+// The provider implementation is built using the [Azure SDK].
+//
+// # Credential Types
+//
+// The supported credential types are:
+//   - service-principal-secret
+//   - managed-identity
+//
+// The recommended way to create a credential with which to bootstrap is
+// to run
+//
+//	juju add-credential azure
```

Comment:
I'm not sure if this belongs in developer documentation, it seems more suitable for user-facing docs?

Reply:
Yes, let's save this for the user docs. (Save a copy and send it to me.)